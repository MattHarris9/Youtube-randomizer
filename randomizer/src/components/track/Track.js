import React from 'react';
import FancyTrack from './FancyTrack';
import SimpleTrack from './SimpleTrack';

const Track = ({ skey, video, selected, playVideo, visualEffectsEnabled }) =>
  visualEffectsEnabled ? (
    <FancyTrack skey={skey} video={video} selected={selected} playVideo={playVideo} />
  ) : (
    <SimpleTrack skey={skey} video={video} selected={selected} playVideo={playVideo} />
  );

export default Track;
