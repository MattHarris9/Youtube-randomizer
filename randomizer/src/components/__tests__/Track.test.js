import React from 'react';
import renderer from 'react-test-renderer';
import Track from '../Track';

const mockFn = jest.fn();

describe('Track', () => {
  test('renders empty track correctly', () => {
    const tree = renderer.create(<Track />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders track  with empty video correctly', () => {
    const props = {
      skey: 1,
      selected: true,
      playVideo: mockFn,
      video: {},
    };
    const tree = renderer.create(<Track {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders full track correctly', () => {
    const props = {
      skey: 1,
      selected: true,
      playVideo: mockFn,
      video: {
        id: 'UExuVVBuX081eUM4MEVLbFd4c2EtUHFZd3c5ak5UbGVWdy4wNEU1MTI4NkZEMzVBN0JF',
        snippet: {
          title: 'The Cat Empire - Song for Elias',
          description: 'damn chill muzic',
          thumbnails: {
            default: {
              url: 'https://i.ytimg.com/vi/6Fot9g9YKyw/default.jpg',
              width: 120,
              height: 90,
            },
            medium: {
              url: 'https://i.ytimg.com/vi/6Fot9g9YKyw/mqdefault.jpg',
              width: 320,
              height: 180,
            },
            high: {
              url: 'https://i.ytimg.com/vi/6Fot9g9YKyw/hqdefault.jpg',
              width: 480,
              height: 360,
            },
          },
        },
      },
    };
    const tree = renderer.create(<Track {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
