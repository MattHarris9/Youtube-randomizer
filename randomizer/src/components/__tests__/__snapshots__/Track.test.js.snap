// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track renders empty track correctly 1`] = `
Array [
  <li
    className="MuiListItem-root WithStyles(ForwardRef(ListItem))-root-1 MuiListItem-gutters"
    disabled={false}
    id={null}
    onClick={[Function]}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      />
    </div>
    <div
      className="MuiListItemText-root"
    />
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;

exports[`Track renders full track correctly 1`] = `
Array [
  <li
    className="MuiListItem-root WithStyles(ForwardRef(ListItem))-root-1 MuiListItem-gutters Mui-selected"
    data-key={1}
    disabled={false}
    id="UExuVVBuX081eUM4MEVLbFd4c2EtUHFZd3c5ak5UbGVWdy4wNEU1MTI4NkZEMzVBN0JF"
    onClick={[Function]}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle"
      >
        <img
          alt="The Cat Empire - Song for Elias"
          className="MuiAvatar-img"
          src="https://i.ytimg.com/vi/6Fot9g9YKyw/default.jpg"
        />
      </div>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        The Cat Empire - Song for Elias
      </span>
    </div>
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;

exports[`Track renders track  with empty video correctly 1`] = `
Array [
  <li
    className="MuiListItem-root WithStyles(ForwardRef(ListItem))-root-1 MuiListItem-gutters Mui-selected"
    data-key={1}
    disabled={false}
    id={null}
    onClick={[Function]}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      />
    </div>
    <div
      className="MuiListItemText-root"
    />
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
